packages/flutter/lib/src/material/feedback.dart

```diff
@@ -86,6 +87,7 @@ class Feedback {
   ///  * [wrapForTap] to trigger platform-specific feedback before executing a
   ///    [GestureTapCallback].
   static Future<Null> forTap(BuildContext context) async {
+    SemanticsService.tap();
```

Comment:
From the BuildContext you could figure out the renderObject and form there you could probably get to the ID of the semantics node that this tap event is for. If you include that ID in the SemanticsEvent, can we then use that ID on the engine side instead of the id of the currently focused node? The focus might have changed between triggering the tap and processing the tap event. This ID would be stable.

Reply:
This only works if the RenderObject is a semantics boundary, otherwise the actually RenderObject with an id may be an arbitrary child or parent of the node in question.

Reply:
Semantics information always flow up in the tree (with the exception of maybe traversal order) and they can't really "jump over" a semantics node. So, is it not always the first ancestor renderObject with a semantic node that you're looking for?

Reply:
I have a somewhat hacky solution - it requires accessing the private _semantics member of RenderObject though, I don't see a way around this.